score-lang-compiler/src/main/java/org/openscore/lang/compiler/utils/ExecutionStepFactory.java

```diff
@@ -81,12 +82,27 @@ public ExecutionStep createStartStep(Long index, Map<String, Serializable> preEx
     public ExecutionStep createActionStep(Long index, Map<String, Serializable> actionRawData) {
         Validate.notNull(actionRawData, "actionData is null");
         Map<String, Serializable> actionData = new HashMap<>();
+        ActionType actionType;
+
         @SuppressWarnings("unchecked") Map<String, String> javaActionData = (Map<String, String>) actionRawData.remove(SlangTextualKeys.JAVA_ACTION);
-        ActionType actionType = ActionType.PYTHON;
-        if (MapUtils.isNotEmpty(javaActionData)) {
+        @SuppressWarnings("unchecked") String pythonScript = (String) actionRawData.get(ScoreLangConstants.PYTHON_SCRIPT_KEY);
+        boolean javaActionFound = MapUtils.isNotEmpty(javaActionData);
+        boolean pythonScriptFound = StringUtils.isNotEmpty(pythonScript);
+
+        if (javaActionFound && pythonScriptFound) {
+            throw new RuntimeException("Both java action and python script found in action data");
+        }
+
+        if (javaActionFound) {
             actionType = ActionType.JAVA;
             actionData.putAll(javaActionData);
+        } else  if (pythonScriptFound) {
+            actionType = ActionType.PYTHON;
+        } else {
+            // java action or python script data is missing
+            throw new RuntimeException("Invalid action data");
         }
+
         actionData.putAll(actionRawData);
```

Comment:
It seems to me that you should put the action raw data only in the case we are in Python action. Otherwise, you already put the data in the *if java action* section above